### PR TITLE
feat: Rehash exported types

### PR DESCRIPTION
**Pull request by max-stytch**
- Standardizes all request and response types to `$Vertical$Product$Method(Request|Response)` format
- Exposes all request and response types
- Exposes numerous helper types (`Member`, `Organization`, `User`, `Session`, etc..)
- adds a `types` export to `package.json`

### CODE REVIEW

File: lib/b2b/discovery.ts [hunk 1]
```diff
@@ -57,22 +57,27 @@ class Organizations {
...
-    data: DiscoveryOrganizationCreateRequest
-  ): Promise<DiscoveryOrganizationCreateResponse> {
-    return request<DiscoveryOrganizationCreateResponse>(this.fetchConfig, {
-      method: "POST",
-      url: "discovery/organizations/create",
-      data,
-    });
+    data: B2BDiscoveryDiscoveryOrganizationCreateRequest
+  ): Promise<B2BDiscoveryDiscoveryOrganizationCreateResponse> {
+    return request<B2BDiscoveryDiscoveryOrganizationCreateResponse>(
```
Comment (taronish-stytch) [inline lib/b2b/discovery.ts:73]: Is this double `Discovery` correct? In this case would $Method be DiscoveryOrganizationCreate?

**Review by taronish-stytch — APPROVED** [review]
Nice big step in the right direction. Any thoughts on having the vertical be put at the end to help searchability/autocompletion?

File: lib/b2b/discovery.ts (hunk 1 above)
Comment (max-stytch) [reply lib/b2b/discovery.ts:73]: Good catch - with ~160 renames I was bound to mess a few up :)

**Review by edwin-stytch — APPROVED** [review]
lgtm

**Comment by max-stytch** [comment]
> Any thoughts on having the vertical be put at the end to help searchability/autocompletion?

My concern with this is that it makes it very easy to autocomplete to the wrong thing - if I type `SessionsAuthent` it could autocomplete to either the B2B or B2C version.

**Pull request merged**